File: BPMOPMobile_VietNamAirLines/BPMOPMobile_VietNamAirLines/BPMOPMobile.Droid.Core/Component/ComponentRow1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.Graphics;
using Android.OS;
using Android.Runtime;
using Android.Support.V4.Content;
using Android.Util;
using Android.Views;
using Android.Widget;
using BPMOPMobile.Bean;
using BPMOPMobile.Droid.Core.Class;

namespace BPMOPMobile.Droid.Core.Component
{
    public class ComponentRow1 : ComponentBase
    {
        public ViewElement _element { get; set; }
        public LinearLayout _parentView { get; set; }
        public Activity _mainAct;
        public ControlBase control;
        public int _widthScreenTablet = -1;
        public int _flagView = -1;
        public bool _visibleItemLine; // true = hiện Line từng item, false = hiện line của nguyên row
        public ComponentRow1(Activity _mainAct, LinearLayout _parentView, ViewElement _element, int _widthScreenTablet, bool _visibleItemLine, int _flagView)
        {
            this._mainAct = _mainAct;
            this._parentView = _parentView;
            this._element = _element;
            this._widthScreenTablet = _widthScreenTablet;
            this._visibleItemLine = _visibleItemLine;
            this._flagView = _flagView;
            InitializeComponent();
        }
        public override void InitializeComponent()
        {
            switch (_element.DataType)
            {
                case "selectuser":
                case "selectusergroup":
                    control = new ControlSelectUser(_mainAct, _parentView, _element);
                    break;
                case "selectusermulti":
                case "selectusergroupmulti":
                    control = new ControlMultiSelectUser(_mainAct, _parentView, _element);
                    break;
                case "date":
                case "datetime":
                case "time":
                    control = new ControlDate(_mainAct, _parentView, _element);
                    break;
                case "singlechoice":
                    control = new ControlSingleChoice(_mainAct, _parentView, _element);
                    break;
                case "multiplechoice":
                    control = new ControlMultiChoice(_mainAct, _parentView, _element);
                    break;
                case "singlelookup":
                    control = new ControlSingleLookup(_mainAct, _parentView, _element);
                    break;
                case "multiplelookup":
                    control = new ControlMultiLookup(_mainAct, _parentView, _element);
                    break;
                case "number":
                    control = new ControlNumber(_mainAct, _parentView, _element);
                    break;
                case "tabs":
                    control = new ControlTabs(_mainAct, _parentView, _element, _widthScreenTablet);
                    break;
                case "attachment":
                    control = new ControlAttachment(_mainAct, _parentView, _element, _widthScreenTablet);
                    break;
                case "attachmentvertical":
                    control = new ControlAttachmentVertical(_mainAct, _parentView, _element);
                    break;
                case "yesno":
                    control = new ControlYesNo(_mainAct, _parentView, _element);
                    break;
                case "tree":
                    control = new ControlTree(_mainAct, _parentView, _element);
                    break;
                case "attachmentverticalformframe": // chưa có
                    //control = new ControlAttachmentVerticalWithFormFrame(parentView, element, currentIndexPath);
                    break;
                case "textinput":
                    control = new ControlTextInput(_mainAct, _parentView, _element);
                    break;
                case "textinputmultiline":
                    control = new ControlTextInputMultiLine(_mainAct, _parentView, _element);
                    break;
                case "textinputformat": // Text Editor
                    control = new ControlTextInputFormat(_mainAct, _parentView, _element);
                    break;
                case "inputattachmenthorizon":
                //control = new ControlInputAttachmentHorizontal(_mainAct, _parentView, _element, _widthScreenTablet);
                //break;
                case "inputattachmentvertical": // testting
                    control = new ControlInputAttachmentVertical(_mainAct, _parentView, _element, _widthScreenTablet, (int)EnumFormControlView.FlagViewControlAttachment.DetailWorkflow);
                    break;
                case "inputworkrelated": // quy trình liên kết
                    control = new ControlInputWorkRelated(_mainAct, _parentView, _element, _widthScreenTablet);
                    break;
                case "inputgriddetails":
                    control = new ControlInputGridDetails(_mainAct, _parentView, _element, _flagView);
                    break;
                default:
                    control = new ControlText(_mainAct, _parentView, _element);
                    break;
            }
        }
        public override void InitializeFrameView(LinearLayout frame)
        {
            if (control == null)
                return;

            base.InitializeFrameView(frame);
            control.InitializeFrameView(frame);

            if (_visibleItemLine == false) // Ẩn Line của Item -> Load Line của nguyên Row
            {
                control._lnLine.Visibility = ViewStates.Gone;
            }
            else
            {
                control._lnLine.Visibility = ViewStates.Visible;
            }
        }
        public override void SetTitle()
        {
            if (control == null)
                return;

            base.SetTitle();
            control.SetTitle();
        }
        public override void SetValue()
        {
            if (control == null)
                return;

            base.SetValue();
            control.SetValue();
        }
        public override void SetProprety()
        {
            if (control == null)
                return;

            base.SetProprety();
            control.SetProprety();
        }
        public override void SetEnable()
        {
            if (control == null)
                return;

            base.SetEnable();
            control.SetEnable();
        }
    }
}

File: VNABPMOP/app/src/main/java/com/vuthao/bpmop/core/component/ComponentRow1.java
package com.vuthao.bpmop.core.component;

import android.app.Activity;
import android.view.View;
import android.widget.LinearLayout;

import com.vuthao.bpmop.base.Functions;
import com.vuthao.bpmop.base.model.dynamic.ViewElement;
import com.vuthao.bpmop.base.vars.VarsControl;
import com.vuthao.bpmop.core.Vars;

public class ComponentRow1 extends ComponentBase {
    protected ViewElement element;
    protected LinearLayout parentView;
    protected Activity mainAct;
    protected ControlBase control;
    protected int widthScreenTablet = -1;
    protected int flagView = -1;
    protected boolean visibleItemLine; // true = hiện Line từng item, false = hiện line của nguyên row

    public ComponentRow1() {

    }

    public ComponentRow1(Activity mainAct, LinearLayout parentView, ViewElement element, int widthScreenTablet, boolean visibleItemLine, int flagView) {
        this.mainAct = mainAct;
        this.parentView = parentView;
        this.element = element;
        this.widthScreenTablet = widthScreenTablet;
        this.visibleItemLine = visibleItemLine;
        this.flagView = flagView;
        initializeComponent();
    }

    @Override
    public void initializeComponent() {
        super.initializeComponent();
        switch (element.getDataType()) {
            case VarsControl.SELECTUSER:
            case VarsControl.SELECTUSERGROUP: {
                control = new ControlSelectUser(mainAct, parentView, element);
                break;
            }
            case VarsControl.SELECTUSERMULTI:
            case VarsControl.SELECTUSERGROUPMULTI: {
                control = new ControlMultiSelectUser(mainAct, parentView, element);
                break;
            }
            case VarsControl.DATE:
            case VarsControl.DATETIME:
            case VarsControl.TIME: {
                control = new ControlDate(mainAct, parentView, element);
                break;
            }
            case VarsControl.SINGLECHOICE: {
                control = new ControlSingleChoice(mainAct, parentView, element);
                break;
            }
            case VarsControl.MULTIPLECHOICE: {
                control = new ControlMultiChoice(mainAct, parentView, element);
                break;
            }
            case VarsControl.SINGLELOOKUP: {
                control = new ControlSingleLookup(mainAct, parentView, element);
                break;
            }
            case VarsControl.MULTIPLELOOKUP: {
                control = new ControlMultiLookup(mainAct, parentView, element);
                break;
            }
            case VarsControl.NUMBER: {
                control = new ControlNumber(mainAct, parentView, element);
                break;
            }
            case VarsControl.TABS: {
                control = new ControlTabs(mainAct, parentView, element, widthScreenTablet);
                break;
            }
            case VarsControl.ATTACHMENT: {
                control = new ControlAttachment(mainAct, parentView, element, widthScreenTablet);
                break;
            }
            case VarsControl.ATTACHMENTVERTICAL: {
                control = new ControlAttachmentVertical(mainAct, parentView, element);
                break;
            }
            case VarsControl.YESNO: {
                control = new ControlYesNo(mainAct, parentView, element);
                break;
            }
            case VarsControl.TREE: {
                control = new ControlTree(mainAct, parentView, element);
                break;
            }
            // Chưa có
            case VarsControl.ATTACHMENTVERTICALFORMFRAME: {
                break;
            }
            case VarsControl.TEXTINPUT: {
                control = new ControlTextInput(mainAct, parentView, element);
                break;
            }
            case VarsControl.TEXTINPUTMULTILINE: {
                control = new ControlTextInputMultiLine(mainAct, parentView, element);
                break;
            }
            case VarsControl.TEXTINPUTFORMAT: {
                control = new ControlTextInputFormat(mainAct, parentView, element);
                break;
            }
            case VarsControl.INPUTATTACHMENTHORIZON:
            case VarsControl.INPUTATTACHMENTVERTICAL: {
                control = new ControlInputAttachmentVertical(mainAct, parentView, element, widthScreenTablet, (int) Vars.FlagViewControlAttachment.DetailWorkflow);
                break;
            }
            case VarsControl.INPUTWORKRELATED: {
                control = new ControlInputWorkRelated(mainAct, parentView, element, widthScreenTablet);
                break;
            }
            case VarsControl.INPUTGRIDDETAILS: {
                if (!Functions.isNullOrEmpty(element.getDataSource())) {
                    control = new ControlInputGridDetails(mainAct, parentView, element, flagView);
                }
                break;
            }
            default:
                control = new ControlText(mainAct, parentView, element);
                break;
        }
    }

    @Override
    public void initializeFrameView(LinearLayout frame) {
        if (control == null) return;
        super.initializeFrameView(frame);

        control.initializeFrameView(frame);

        if (!visibleItemLine) {
            control.lnLine.setVisibility(View.GONE);
        } else {
            control.lnLine.setVisibility(View.VISIBLE);
        }
    }

    @Override
    public void setTitle() {
        if (control == null) return;
        super.setTitle();

        control.setTitle();
    }

    @Override
    public void setValue() {
        if (control == null) return;
        super.setValue();

        control.setValue();
    }

    @Override
    public void setProprety() {
        if (control == null) return;
        super.setProprety();

        control.setProprety();
    }

    @Override
    public void setEnable() {
        if (control == null) return;
        super.setEnable();

        control.setEnable();
    }
}
